src-docs/src/views/icon/icons.js

```diff
@@ -279,6 +279,14 @@ export const iconTypes = [
   'wrench',
 ];
 
+// Add mapping icon name for backwards compatibility
+const normalizeIconType = (iconType) => {
+  const mapping = {
+    crossInACircleFilled: 'crossInCircleFilled',
+  };
+  return mapping[iconType] || iconType;
+};
+
```

Comment:
@Willie-The-Lord Revert this function and its usage, as it's not needed here. What you want to do instead is replace `crossInACircleFilled`with `crossInCircleFilled` in `iconTypes` at L74.

It's important to understand how this `iconList` is used and why it's different from the `typeToPathMap` in `src/components/icon`. The list in this file is icons that will appear in the documentation website in the icon list: https://oui.opensearch.org/1.3/#/display/icons

So, because we only want the "correct" spelling of the icon to appear in the docs, we don't want to include the mispelling in that list. Note that this list _also_ is used by the doc-site-only "iconValidator" function: https://github.com/opensearch-project/oui/blob/main/src-docs/src/services/playground/iconValidator.js

By leaving the misspelled version off this list, when users try it in a playground component on the docs site, the validator will block it as an undocumented icon type, which is actually a nice subtle hint. However, if you actually use the old misspelled version as an icon type, it will work anywhere outside the doc site (because we don't use that validator, we use the actual types from the mapping. To see what I mean, after changing this file, you can try it with the `iconType` property in the playground tab of http://localhost:8030/#/display/badge/playground.

Basically, the list in this file is not all possible iconType strings, but the list that we encourage developers to use.

Reply:
> However, if you actually use the old misspelled version as an icon type, it will work anywhere outside the doc site (because we don't use that validator, we use the actual types from the mapping.)

Ahh! That totally makes sense! I kinda mixed up the concept of the `iconType` and its usage. Now everything is making sense! Thanks for the clarification. By the way, the playground tip actually makes everything even clearer. Thanks, @joshuarrrr.